plugins/webp-uploads/tests/test-picture-element.php

```diff
@@ -115,32 +111,68 @@ public function test_maybe_wrap_images_in_picture_element( bool $jpeg_and_webp,
+				'fallback_jpeg'   => false,
+				'picture_element' => true,
+				'expected_html'   => '<img width="{{img-width}}" height="{{img-height}}" src="{{img-src}}" class="wp-image-{{img-attachment-id}}" alt="{{img-alt}}" decoding="async" loading="lazy" srcset="{{img-srcset}}" sizes="{{img-sizes}}" />',
 			),
 			'only jpeg enabled'        => array(
-				'jpeg_and_webp'          => true,
-				'picture_element'        => false,
-				'expect_picture_element' => false,
-				'expected_html'          => '<img width="{{img-width}}" height="{{img-height}}" src="{{img-src}}" class="wp-image-{{img-attachment-id}}" alt="{{img-alt}}" decoding="async" loading="lazy" srcset="{{img-srcset}}" sizes="{{img-sizes}}" />',
+				'fallback_jpeg'   => true,
+				'picture_element' => false,
+				'expected_html'   => '<img width="{{img-width}}" height="{{img-height}}" src="{{img-src}}" class="wp-image-{{img-attachment-id}}" alt="{{img-alt}}" decoding="async" loading="lazy" srcset="{{img-srcset}}" sizes="{{img-sizes}}" />',
 			),
 			'neither enabled'          => array(
-				'jpeg_and_webp'          => false,
-				'picture_element'        => false,
-				'expect_picture_element' => false,
-				'expected_html'          => '<img width="{{img-width}}" height="{{img-height}}" src="{{img-src}}" class="wp-image-{{img-attachment-id}}" alt="{{img-alt}}" decoding="async" loading="lazy" srcset="{{img-srcset}}" sizes="{{img-sizes}}" />',
+				'fallback_jpeg'   => false,
+				'picture_element' => false,
+				'expected_html'   => '<img width="{{img-width}}" height="{{img-height}}" src="{{img-src}}" class="wp-image-{{img-attachment-id}}" alt="{{img-alt}}" decoding="async" loading="lazy" srcset="{{img-srcset}}" sizes="{{img-sizes}}" />',
 			),
 		);
 	}
 
+	public function test_picture_source_only_have_additional_mime_not_jpeg_and_return_jpeg_fallback(): void {
+		$mime_type = 'image/webp';
+		if ( ! wp_image_editor_supports( array( 'mime_type' => $mime_type ) ) ) {
+			$this->markTestSkipped( "Mime type $mime_type is not supported." );
+		}
+
+		// Create some content with the image.
+		$image = wp_get_attachment_image(
+			self::$image_id,
+			'large',
+			false,
+			array(
+				'class' => 'wp-image-' . self::$image_id,
+				'alt'   => 'Green Leaves',
+			)
+		);
+
+		$img_markup = apply_filters( 'the_content', $image );
+
+		$img_processor = new WP_HTML_Tag_Processor( $img_markup );
+		$this->assertTrue( $img_processor->next_tag( array( 'tag_name' => 'IMG' ) ), 'There should be an IMG tag.' );
+		$img_src    = $img_processor->get_attribute( 'src' );
+		$img_srcset = $img_processor->get_attribute( 'srcset' );
```

Comment:
Let's make sure these are actually WebP images here.

```suggestion
		$img_srcset = $img_processor->get_attribute( 'srcset' );
		$this->assertStringEndsWith( '.webp', $img_src );
		$this->assertStringContainsString( '.webp', $img_srcset );
```

Reply:
Check my reply https://github.com/WordPress/performance/pull/1408#discussion_r1706364060